cmake/Modules/FindITT.cmake

```diff
@@ -0,0 +1,28 @@
+# - Try to find ITT
+#
+# The following are set after configuration is done:
+#  ITT_FOUND          : set to true if ITT is found.
+#  ITT_INCLUDE_DIR    : path to ITT include dir.
+#  ITT_LIBRARIES      : list of libraries for ITT
+
+IF (NOT ITT_FOUND)
+SET(ITT_FOUND OFF)
+
+SET(ITT_INCLUDE_DIR)
+SET(ITT_LIBRARIES)
+
+IF(NOT CMAKE_BUILD_TYPE)
+  SET(CMAKE_BUILD_TYPE "Release" CACHE STRING
+      "Choose the type of build, options are: Debug Release"
+      FORCE)
+ENDIF(NOT CMAKE_BUILD_TYPE)
+
+SET(ITT_ROOT "${PROJECT_SOURCE_DIR}/third_party/ittapi")
+FIND_PATH(ITT_INCLUDE_DIR ittnotify.h PATHS ${ITT_ROOT} PATH_SUFFIXES include)
+IF (ITT_INCLUDE_DIR)
+  ADD_SUBDIRECTORY(${ITT_ROOT})
+  SET(ITT_LIBRARIES ittnotify)
+  SET(ITT_FOUND ON)
+ENDIF (ITT_INCLUDE_DIR)
```

Comment:
This looks like a really old pattern, can you please use `add_library(ITT` and `target_include_directories` instead?
This way, ITT includes would be applied only to objects depending on it (and this dependency should be private, i.e. those headers do not need to be installed as part of libtorch, do they?)

Reply:
Sure.
No, ITT headers are not necessary to be exposed out. They can be private.

Reply:
HI @malfet , just would like to double confirm with you. ittapi already has a `CMakeLists.txt` containing `add_library` at https://github.com/intel/ittapi/blob/master/CMakeLists.txt. What we do in the `FindITT.cmake` is to invoke this `CMakeLists.txt`. Should we use `add_subdirectory` in the `FindITT.cmake` or `add_library` in this case?